[Team-03] [Android] [3주차 1번째 PR ] 깃허브 로그인 구현, 이슈 조회, Api 조회 및 생성 구현

shncder opened the pull request:
## 진행사항

- 지난 PR 리뷰 반영

  - Dialog을 네비게이팅으로 연결했습니다.
  - Issue의 checkBox는 viewModel에서 관리하되 뷰의 ui는 adapter에서 구현하도록 유지했습니다.
  - label의 checkBox는 viewModel에서 관리하되 뷰의 ui는 fragment에서 구현하도록 유지했습니다.
 
 - 이슈 아이템
   - EditModel 상태 변수 viewModel로 이동

 - 라벨 아이템
   - EditModel 상태 변수 viewModel로 이동
   - Item checkBox Ui 변수 fragment로 이동

- 이슈 디테일 화면
  - DetailAdapter의 getViewType 분기문 수정
  - DetailDialog 이동함수 -> navigate로 변경

- 로그인 화면
  - Hilt Di 적용
  - Redirect된 페이지에서 html 소스 파싱하는 로직 추가
  - SharedPreference 관리 클래스 선언
  - JWT 토큰을 AccessToken.token에 저장하는 로직까지 테스트 완료

- 이슈 Api
  - IssueRepostiory, DataSource, api 구현

- 라벨 Api
  - Label Api, Dto, Datasource 정의
  - 생성시 보낼 데이터를 위한 LabelRequestDto 정의
  - DI 수정

## 추후 진행사항
 - Api 추가구현

## 결과
![res5](https://user-images.githubusercontent.com/91953080/176385950-4a4fe938-1475-4359-9699-c785576f9197.gif)


## 질문사항
 - editMode 상태변수를 viewModel에서 관리하면서,   라벨의 checkbox 로직은 어댑터에 callBack을 넘겨줌으로써 구현하고,  Issue는 Model에 edtiable 멤버변수를 추가해주는 방향으로 구현해보았는데  어떤 방향성이 더 나은지 궁금합니다

- 프로그래스바말고 취업사이트 좋은곳이 있는지 궁금합니다.

kang-proto committed:
fix:  두번째주차 금요일 PR 리뷰 반영

- DetailAdapter의 getViewType 분기문 수정
- DetailDialog 이동함수 -> navigate로 변경
- EditModel 상태 변수 viewModel로 이동

shncder committed:
feat: label checkBox 상태 저장장소 변경

- checkBox 상태 저장장소 변경중

HyungMinKang committed:
Merge pull request #43 from naneun/fix/SecondWeekSecondFeedBack

[Android][히데] 두번째주차 금요일 PR 리뷰 반영

shncder committed:
feat: label 'clLabelSelector' Fragment에서 관리

- checkBox 상태 저장장소 변경중

kang-proto committed:
feat: 깃허브 로그인 구현

- Hilt Di 적용
- Redirect된 페이지에서 html 소스 파싱하는 로직 추가
- SharedPreference 관리 클래스 선언
- JWT 토큰을 AccessToken.token에 저장하는 로직까지 테스트 완료

kang-proto committed:
feat: 이슈 리스트 받아오기 구현

- IssueRepostiory, DataSource, api 구현

shncder committed:
feat: label item checkBox 구현

- label item의 Adapter에서 checkbox를 꺼내 Fragment에서 관리하도록 수정

shncder committed:
Merge pull request #44 from naneun/feature/gitHubLogin

[Android][히데] Hilt DI적용,   깃허브 로그인 ,  이슈 리스트 api 적용

shncder committed:
chore : merge 및 불필요한 import 제거

- merge 작업
- 불필요한 import 제거

HyungMinKang committed:
Merge pull request #45 from naneun/feature/checkBox

[Android][JungPark] Label의 Edit모드 구현

kang-proto committed:
feat: 라벨 조회, 생성 api 구현

- Label Api, Dto, Datasource 정의
- 생성시 보낼 데이터를 위한 LabelRequestDto 정의
- DI 수정

shncder committed:
Merge pull request #46 from naneun/feature/labelListApi

[Android][히데] 라벨 조회, 생성  구현

renovatio0424 commented:
# 질문 1
>editMode 상태변수를 viewModel에서 관리하면서, 라벨의 checkbox 로직은 어댑터에 callBack을 넘겨줌으로써 구현하고, Issue는 Model에 edtiable 멤버변수를 추가해주는 방향으로 구현해보았는데 어떤 방향성이 더 나은지 궁금합니다

질문이 잘 이해가 안가네요 
어떤 부분이 문제인건가요?
# 질문 2
>프로그래스바말고 취업사이트 좋은곳이 있는지 궁금합니다. 

최대한 많은 채용사이트들을 이용하시면 좋을거 같아요 
특히 신입 개발자의 경우에는 회사를 가려 지원하기 보다는 최대한 많은 회사들 면접을 보면서 경험해보시는게 도움이 될거 같습니다. 

저의 경우에는 거의 가리지 않고 채용사이트를 이용합니다. (개인적으로는 리멤버가 가장 반응이 빠르고 좋았던걸로 기억하는데 신입도 그럴지는 잘모르겠네요)
1. 원티드
2. 로켓펀치
3. 리멤버
4. 프로그래머스
5. 사람인

renovatio0424 commented on Android/app/src/main/java/com/example/issue_tracker/common/AccessToken.kt:
```diff
@@ -0,0 +1,7 @@
+package com.example.issue_tracker.common
+
+object AccessToken {
+    var code: String = ""
+    var token: String = ""
+
+}
```
`AccessToken` 을 `object` 객체로 만든 의도가 무엇인가요?

HyungMinKang replied:
앱 전역에서 App LIfeCycle과 동일하게 해당 값들이 유지되야 한다고 생각해서 object 객체로 만들었습니다.

renovatio0424 commented on Android/app/src/main/java/com/example/issue_tracker/ui/common/SharedPreferenceManager.kt:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    fun putString(key:String, value:String){
+        sharedPreferences.edit().putString(key,value).apply()
```
`sharedPreferences` 가 초기화되지 않은 경우에 호출할 경우 에러가 발생하지 않을까요? 
좀 위험한 코드 같네요

HyungMinKang replied:
sharedPreference가 null이 아닐때만  작동하도록 수정할 예정입니다

renovatio0424 commented on Android/app/src/main/java/com/example/issue_tracker/ui/label/LabelFragment.kt:
```diff
@@ -1,67 +1,141 @@
-        binding.iBtnLabelAdd.setOnClickListener {
-            navigator.navigate(R.id.action_navigation_label_to_labelWriteFragment)
+    private fun addCheckList(itemId: Int) {
+        viewLifecycleOwner.lifecycleScope.launch {
+            viewModel.addCheckList(itemId)
+            viewModel.labelEditModeOn()
+            setSelectedIssueCount()
+        }
```
이 동작들을 굳이 코루틴으로 만든 이유가 있나요? 
그리고 `addCheckList` 가 호출되었을때에 대한 처리를 `viewModel` 안에 리하면 되지 않을까요? 
`View` 에서 `ViewModel` 을 핸들링하는건 나중에 문제가 생길 여지가 있어보이네요

renovatio0424 commented on Android/app/src/main/java/com/example/issue_tracker/ui/login/LogInActivity.kt:
```diff
@@ -1,26 +1,100 @@
+
+
+    }
+    internal class MyJavaScriptInterface {
```
네이밍이 좀 ㅎㅎ;
`LoginJavaScriptInterface` 정도가 어떨지 ㅎㅎ

HyungMinKang replied:
넵 수정하겠습니다

kang-proto committed:
feat: 라벨 삭제 기능 구현

- 체크박스로 선택된 id를 List로 서버측에 보냄
- 삭제후 리스트화면으로 돌아오면서 재로딩

kang-proto committed:
feat: 로그인한 계정정보 페이지 구현, 비로그인 모드 구현

- 로그인 했을시 내계정 페이지 이동시 정보 출력
- 로그인 하지않았을시 로그인이 필요한 기능이라느 화면 보여줌
- 앱에 등록되어있는 회원 리스트 받아오기 (User Api, User DataSource, UserRemoteDataSource)
- Di Module 업데이트
- SharedPreferenceManager에  sharedPreference null 체크 로직 추가

kang-proto committed:
feat: 이슈 필터 처리 구현

- 초기화, 적용 가능 여부에 따라 enable true/false로 처리
- spinner에 itemSelect 내부 분기문으로 선택된 아이템 viewModel에 저장
- api를 통해 필터 적용 이슈 리스트 리턴

kang-proto committed:
feat:  이미지 파일 서버에 업로드 구현,  이슈 등록 구현

- 앨범에서 선택한 이미지 파일 생성
- MultipartBody.Part를 이용해 서버측에 파일을 보내고 url을 리턴받음
- File -> MultiparBody.Part로 변경하기 위한 object 선언
- Issue 생성 request를 보내기 위한 IssueRequestDto 생성
- DI Update

kang-proto committed:
feat:  이슈 디테일, 코멘트 페이징 처리

- 이슈 디테일 정보를 받아오기 의한 네트워크 처리 구현
- 디테일 화면의 코멘트 부분에 페이징 처리
- ProgressBar 모델 , ViewType 정의
- 리사이클러뷰에 addOnScrollListener 등록

shncder committed:
Merge pull request #47 from naneun/feature/labelListApi

[Android][히데]  Label 생성, 조회, 삭제 Api 처리,  MyAccount 화면 네트워크 처리,  이슈 작성화면에서 이미지 업로드 구현,  이슈 조회, , 삭제, 검색 , 상태변경 api 적용, 이슈 디테일 , 커멘트 조회 api 적용

shncder committed:
feat: MilestoneApi 작업

- Milestone Api, Dto, Datasource 정의
- 생성시 보낼 데이터를 위한 MilestoneRequestDto 정의
- 생성후 리스트화면으로 돌아오면서 재로딩

HyungMinKang committed:
Merge branch 'Android' into feautre/milestoneApi

HyungMinKang committed:
Merge pull request #48 from naneun/feautre/milestoneApi

[Android][JungPark] MilestoneApi 작업
